zenodo/modules/spam/utils.py

```diff
@@ -26,32 +26,83 @@
 
 from __future__ import absolute_import, print_function
 
-from flask import current_app, render_template
+from elasticsearch_dsl import Q
+from flask import current_app, flash, render_template
 from flask_babelex import gettext as _
 from flask_mail import Message
+from flask_principal import ActionNeed
+from invenio_access import Permission
+from invenio_communities.models import Community
 from invenio_mail.tasks import send_email
+from invenio_search.api import RecordsSearch
+from werkzeug.exceptions import HTTPException
 
+from zenodo.modules.spam.tasks import check_metadata_for_spam
 
-def send_spam_user_email(recipient):
+
+def send_spam_user_email(recipient, deposit=None, community=None):
     """Send email notification to blocked user after spam detection."""
     msg = Message(
-        _("Your Zenodo Upload has been automatically marked as Spam."),
+        _("Your Zenodo activity has been automatically marked as Spam."),
         sender=current_app.config.get('SUPPORT_EMAIL'),
         recipients=[recipient],
     )
-    msg.body = render_template("zenodo_spam/email/spam_user_email.tpl")
+    msg.body = render_template(
+        "zenodo_spam/email/spam_user_email.tpl",
+        community=community,
+        deposit=deposit
+        )
     send_email.delay(msg.__dict__)
 
 
-def send_spam_admin_email(deposit, user):
+def send_spam_admin_email(user, deposit=None, community=None):
     """Send email notification to admins for a spam detection."""
     msg = Message(
-        _("Zenodo Deposit Marked as Spam."),
+        _("Zenodo activity marked as spam."),
         sender=current_app.config.get('SUPPORT_EMAIL'),
         recipients=[current_app.config.get('ZENODO_ADMIN_EMAIL')],
     )
-    msg.body = render_template(
-        "zenodo_spam/email/spam_admin_email.tpl",
-        user=user,
-        deposit=deposit)
+    if deposit:
+        msg.body = render_template(
+            "zenodo_spam/email/spam_deposit_admin_email.tpl",
+            user=user,
+            deposit=deposit)
+    if community:
+        msg.body = render_template(
+            "zenodo_spam/email/spam_community_admin_email.tpl",
+            user=user,
+            community=community)
     send_email.delay(msg.__dict__)
+
+
+def spam_checking_helper(community=None, deposit=None):
```

Comment:
```suggestion
def check_spam(community=None, deposit=None):
```

If it was an object/class it would be fitting, but having functions as verbs/actions reads a bit better

Reply:
The function though is a wrapper on the task that is actually checking for spam and additionally includes the handling of the spam content so IMHO this name wouldn't be descriptive enough.